src/main/java/org/openrewrite/java/migrate/util/OptionalNotEmptyToIsPresent.java

```diff
@@ -55,29 +55,29 @@ public TreeVisitor<?, ExecutionContext> getVisitor() {
         TreeVisitor<?, ExecutionContext> check = Preconditions.and(
                 new UsesJavaVersion<>(11),
                 new UsesMethod<>(JAVA_UTIL_OPTIONAL_IS_EMPTY));
-        MethodMatcher optionalIsPresentMatcher = new MethodMatcher(JAVA_UTIL_OPTIONAL_IS_EMPTY);
         return Preconditions.check(check, new JavaVisitor<ExecutionContext>() {
             @Override
-            public Statement visitStatement(Statement s, ExecutionContext ctx) {
-                Statement statement = (Statement) super.visitStatement(s, ctx);
-                if (statement instanceof J.Unary) {
-                    J.Unary unary = (J.Unary) statement;
+            public J visitStatement(Statement s, ExecutionContext ctx) {
+                J superReturn = super.visitStatement(s, ctx);
+                if (superReturn instanceof J.Unary) {
+                    J.Unary unary = (J.Unary) superReturn;
                     if (unary.getOperator() == Type.Not) {
                         Expression expression = unary.getExpression();
                         if (expression instanceof J.MethodInvocation) {
                             J.MethodInvocation m = (J.MethodInvocation) expression;
+                            MethodMatcher optionalIsPresentMatcher = new MethodMatcher(JAVA_UTIL_OPTIONAL_IS_EMPTY);
```

Comment:
What's the reason to move creation of this method matcher into the visitor? Now we potentially create a lot more method matchers, whereas before that would have been a shared instance.

Reply:
You are right, I seem to have confused myself there, thinking that having this only after the many if conditions would reduce the number of instances. I clearly missed that recipe is instantiated once, while the visitor is many times.